Objects/exceptions.c

```diff
@@ -629,6 +629,454 @@ ComplexExtendsException(PyExc_BaseException, SystemExit, SystemExit,
                         0, 0, SystemExit_members, 0, 0,
                         "Request to exit from the interpreter.");
 
+/*
+ *    BaseExceptionGroup extends BaseException
+ *    ExceptionGroup extends BaseExceptionGroup and Exception
+ */
+
+PyObject *PyExc_ExceptionGroup;
+
+static inline PyBaseExceptionGroupObject*
+_PyBaseExceptionGroupObject_cast(PyObject *exc)
+{
+    assert(PyObject_TypeCheck(exc, (PyTypeObject *)PyExc_BaseExceptionGroup));
+    return (PyBaseExceptionGroupObject *)exc;
+}
+
+static PyObject *
+BaseExceptionGroup_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+{
+    PyObject *msg = NULL;
+    PyObject *excs = NULL;
+
+    if (!PyArg_ParseTuple(args, "OO", &msg, &excs)) {
+        return NULL;
+    }
+    if (!PyUnicode_CheckExact(msg)) {
+        PyErr_SetString(PyExc_TypeError, "Expected a message");
+        return NULL;
+    }
```

Comment:
(a) Can't we express that with a format code to `PyArg_ParseTuple`?

(b) If not, perhaps clarify the error? It should probably refer to the first argument and state that a string is expected.

Reply:
I could use U but then it can't be None. Should we allow None?

I see that some other exceptions just store whatever you give them for the message and only check if it's unicode in str().   Here this can be confusing - people will forget the message and try ExceptionGroup(exc1, exc2), then it will assume that exc1 is the message and complain that exc2 is not a sequence.

Reply:
Ok, I'm using U now.